src/core/meshcop/dataset.cpp

```diff
@@ -251,6 +244,8 @@ otError Dataset::Set(const Dataset &aDataset)
         Remove(Tlv::kDelayTimer);
     }
 
+    mUpdateTime = aDataset.GetUpdateTime();
```

Comment:
Could we use `Timer::GetNow()` here to indicate when the local dataset has a update by copying the entire received dataset?